lib/ansible/plugins/action/synchronize.py

```diff
@@ -155,6 +155,11 @@ def run(self, tmp=None, task_vars=None):
         if task_vars is None:
             task_vars = dict()
 
+        # We make a copy of the args here because we may fail and be asked to
+        # retry. If that happens we don't want to pass the munged args through
+        # to our next invocation. Munged args are single use only.
+        self._task._tmp_args = self._task.args.copy()
```

Comment:
we should already be inside a copy of the task, not sure why we would need to copy args again.